src/installers.ts

```diff
@@ -1665,156 +1635,289 @@ const installers: Installer[] = [
+  // since we do have a reasonable indication the
+  // top of the mod is a wrapper
+  return false;
+};
+
+const enum Transform {
+  Unwrapped = `Tree with extra top-level directory removed`,
+  None = `No transforms`,
+}
+
+// This is probably nonsense especially since we're not giving
+// these types the *actual* transform to complete whenever. That's
+// just way overkill for this problem, but result is that this is
+// kind of a half-assed type for nothing more than a bit of safety.
+interface NotModifiedFileTree {
+  readonly transform: Transform.None;
+  readonly fileTree: FileTree;
+}
+
+interface UnwrappedFileTree {
+  readonly transform: Transform.Unwrapped;
+  readonly fileTree: FileTree;
+  readonly originalTree: FileTree;
+  readonly wrapperDir: string;
+}
+
+type ProcessedFileTree = NotModifiedFileTree | UnwrappedFileTree;
+
+const unwrapTreeIfNecessary = (api: VortexApi, fileTree: FileTree): ProcessedFileTree => {
+  const haveGiftwrappedMod = detectGiftwrapLayout(fileTree);
+
+  const wrapperDir = subdirNamesIn(FILETREE_ROOT, fileTree)[0];
+
+  const transformedTree = haveGiftwrappedMod
+    ? subtreeFrom(wrapperDir, fileTree)
+    : fileTree;
+
+  const unwrappedPaths = sourcePaths(transformedTree);
+
+  if (haveGiftwrappedMod) {
+    api.log(`info`, `Mod was giftwrapped, unwrapped it for install.`);
+    api.log(`debug`, `Using unwrapped filetree: `, unwrappedPaths);
+
+    return {
+      transform: Transform.Unwrapped,
+      fileTree: transformedTree,
+      originalTree: fileTree,
+      wrapperDir,
+    } as UnwrappedFileTree;
+  }
+
+  return {
+    transform: undefined,
+    fileTree,
+  } as NotModifiedFileTree;
+};
+
+const giftwrapSourcesIfNecessary = (
+  api: VortexApi,
+  treeInUse: ProcessedFileTree,
+  instructions: VortexInstruction[],
+): VortexInstruction[] =>
+  treeInUse.transform === Transform.Unwrapped
+    ? instructions.map(({ type, source, destination }: VortexInstruction) => ({
+        type,
+        source: path.join(treeInUse.wrapperDir, source),
+        destination,
+      }))
+    : instructions;
+
+//
+// (wrap) `testSupported`
+//
+//  Before hitting actual installers(s), make sure the tree is the way
+//  we want it to be - no extra toplevel dir for example.
+//
+//  Also passes in some extra data like `VortexApi` and our `FileTree`.
+//
+//  @type `VortexTestSupportedFunc`
+//
 export const wrapTestSupported =
   (
     vortex: VortexExtensionContext,
     vortexApiThing,
     installer: Installer,
   ): VortexTestSupportedFunc =>
-  (files: string[], gameId: string, ...args) => {
+  (files: string[], gameId: string) => {
+    //
     const vortexLog: VortexLogFunc = vortexApiThing.log;
-    const vortexApi: VortexApi = {
-      ...vortex.api,
-      log: vortexApiThing.log,
-    };
+    const vortexApi: VortexApi = { ...vortex.api, log: vortexApiThing.log };
 
     if (gameId !== GAME_ID) {
+      vortexApi.log(`error`, `Not a ${GAME_ID} mod: ${gameId}`);
```

Comment:
Don't think an error log is needed here.. It is pretty standard for it to not be a mod for a given game